gluster-exporter/metric_pv_lv_vg.go

```diff
@@ -0,0 +1,114 @@
+package main
+
+import (
+	"bufio"
+	"bytes"
+	"os/exec"
+	"strings"
+
+	"github.com/prometheus/client_golang/prometheus"
+	log "github.com/sirupsen/logrus"
+)
+
+type collectDetails struct {
+	cmdStr string
+}
+
+// executes the command, and collect the details
+// returns the array of non empty output lines
+func (cd collectDetails) ExecCommand() ([]string, error) {
+	var outLines []string
+	cmdArr := strings.Fields(strings.TrimSpace(cd.cmdStr))
+	mainCmd, err := exec.LookPath(cmdArr[0])
+	if err != nil {
+		mainCmd = cmdArr[0]
+	}
+	outB, err := exec.Command(mainCmd, cmdArr[1:]...).Output()
+	if err != nil {
+		return nil, err
+	}
+	outR := bytes.NewReader(outB)
+	outScnr := bufio.NewScanner(outR)
+	// split each line
+	outScnr.Split(bufio.ScanLines)
+	log.Print("Output\n\n")
+	noOfLines := 0
+	for outScnr.Scan() {
+		line := strings.TrimSpace(outScnr.Text())
+		if line == "" {
+			continue
+		}
+		outLines = append(outLines, line)
+		noOfLines++
+	}
+	log.Println("No: of lines:", noOfLines)
+	return outLines, err
+}
+
+var (
+	metricLabels = []MetricLabel{
+		{
+			Name: "name",
+			Help: "Name of the metric, whose count is collected (PV or LV or VG etc)",
+		},
+	}
+
+	glusterPVCountMetric = newPrometheusGaugeVec(Metric{
```

Comment:
Lets add another metric called `gluster_thin_pool_count` as well which you vcan figure out from `lvs` command I feel.
The labels for this metric would be `peerid` and `vg_name`.